migtests/scripts/live-run-test.sh

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+
+set -e
+set -m
+
+if [ $# -ne 1 ]
+then
+	echo "Usage: $0 TEST_NAME"
+	exit 1
+fi
+
+set -x
+
+export TEST_NAME=$1
+
+export REPO_ROOT="${PWD}"
+export SCRIPTS="${REPO_ROOT}/migtests/scripts"
+export TESTS_DIR="${REPO_ROOT}/migtests/tests"
+export TEST_DIR="${TESTS_DIR}/${TEST_NAME}"
+export EXPORT_DIR=${EXPORT_DIR:-"${TEST_DIR}/export-dir"}
+
+export PYTHONPATH="${REPO_ROOT}/migtests/lib"
+
+# Order of env.sh import matters.
+source ${TEST_DIR}/env.sh
+
+if [ "${SOURCE_DB_TYPE}" = "oracle" ]
+then
+	source ${SCRIPTS}/${SOURCE_DB_TYPE}/live_env.sh 
+else
+	source ${SCRIPTS}/${SOURCE_DB_TYPE}/env.sh
+fi
+
+source ${SCRIPTS}/yugabytedb/env.sh
+
+source ${SCRIPTS}/functions.sh
+
+main() {
+
+	echo "Deleting the parent export-dir present in the test directory"
+	rm -rf ${EXPORT_DIR}	
+	echo "Creating export-dir in the parent test directory"
+	mkdir -p ${EXPORT_DIR}
+	echo "Assigning permissions to the export-dir to execute init-db, cleanup-db scripts"
+	chmod +x ${TEST_DIR}/init-db ${TEST_DIR}/cleanup-db
+
+	step "START: ${TEST_NAME}"
+	print_env
+
+	pushd ${TEST_DIR}
+
+	step "Initialise source database."
+	./init-db
+
+	step "Grant source database user permissions"
+	if [ "${SOURCE_DB_TYPE}" = "oracle" ]
+	then
+		grant_live_user_permission_oracle ${ORACLE_CDB_NAME} ${SOURCE_DB_NAME}
+	else
+		grant_permissions ${SOURCE_DB_NAME} ${SOURCE_DB_TYPE} ${SOURCE_DB_SCHEMA}
+	fi
+	
+	step "Export schema."
+	export_schema
+	find ${EXPORT_DIR}/schema -name '*.sql' -printf "'%p'\n"| xargs grep -wh CREATE
+
+	step "Analyze schema."
+	analyze_schema
+	tail -20 ${EXPORT_DIR}/reports/report.txt
+
+	step "Fix schema."
+	if [ -x "${TEST_DIR}/fix-schema" ]
+	then
+		 "${TEST_DIR}/fix-schema"
+	fi
+
+	step "Analyze schema."
+	analyze_schema
+	tail -20 ${EXPORT_DIR}/reports/report.txt
+
+	step "Create target database."
+	run_ysql yugabyte "DROP DATABASE IF EXISTS ${TARGET_DB_NAME};"
+	run_ysql yugabyte "CREATE DATABASE ${TARGET_DB_NAME}"
+
+	step "Import schema."
+	import_schema
+	run_ysql ${TARGET_DB_NAME} "\dt"
+
+	step "Export data."
+	# false if exit code of export_data is non-zero
+	export_data --live-migration || { 
+		tail_log_file "yb-voyager.log"
+		tail_log_file "debezium.log"
+		exit 1
+	} &
+
+	# Storing the pid for the export data command
+	exp_pid=$!
+
+	# Killing the export process in case of failure
```

Comment:
What does this exactly do? If the export_data dies, this run-test process will also die?

Reply:
Quite the opposite, I observed that when the run-test process died/errored out/failed validation, the export was still running. So this helps to stop the export in such cases.

Reply:
Got it 👍